### DISCUSSION TITLE
mpegts & klv-metadata to rtsp?

### DISCUSSION THREAD

**Discussion by Hippakaveri** [bluenviron/mediamtx#2610] [Questions]
### Question

Does mediamtx support klv-metadata in rtsp? I've got an udp source, that provides mpegts & klv-metadata.
If I just use source configuration, it will just take the video stream and leaves the klv-stream out of it. (Verified by ffprobe)

Below source information checked with ffprobe.
"Input #0, mpegts, from 'udp://(multicastaddr):(port)':
  Duration: N/A, start: 20.670000, bitrate: N/A
  Program 1
    Metadata:
      service_name    : Service01
      service_provider: FFmpeg
  Stream #0:0[0x100]: Video: h264 (High) ([27][0][0][0] / 0x001B), yuv420p(progressive), 1280x720, 29.92 tbr, 90k tbn, 180k tbc
  Stream #0:1[0xfc]: Data: klv (KLVA / 0x41564C4B)"

All though there seems to be some kind of support for it? As by using gstreamer I've managed to split the source stream to 2 separate 
rtsp paths, and I've managed to validate the data is comming there. But when I try to do the same to include them in the single rtsp-path, I can see both streams there, but no data coming through. 

So should MediaMTX be able to pass both video & metadata coming from the udp source, by default or is there something that needs to be configured? And if not how come does it become visible if passed through gstreamer?

Below those gstreamer commands (which were ran in "runOnInit")

Sends video and klv to separate rtsp paths:
gst-launch-1.0 udpsrc multicast-group=(multicastaddr) port=123456 ! tsdemux name=demux
demux. ! queue ! h264parse ! decodebin ! videorate ! videoscale ! x264enc tune=zerolatency ! mpegtsmux ! rtspclientsink location=rtsp://127.0.0.1:8554/testpath
demux. ! queue ! meta/x-klv ! rtspclientsink location=rtsp://127.0.0.1:8554/metadata

BOTH KLV AND VIDEO IN SAME RTSP PATH, but no data in klv-stream
gst-launch-1.0 udpsrc multicast-group=(multicastaddr) port=123456 ! tsdemux name=demux
demux. ! queue ! h264parse ! decodebin ! videorate ! videoscale ! x264enc tune=zerolatency ! mpegtsmux name=mux
demux. ! queue ! meta/x-klv ! mux.
mux. ! rtspclientsink location=rtsp://127.0.0.1:8554/testpath

Thanks in advance.

**Comment by aler9** [comment]
Hello, this is currently unsupported but it will be in the future, the feature request is here: https://github.com/bluenviron/mediamtx/issues/2693

[a comment by an automated account is not shown]
